src/components/ProductionLine.js

```diff
@@ -0,0 +1,59 @@
+import React, { useReducer } from "react";
+import { ProductionLineContainer, Main } from "./styled";
+import { initialState, reducer } from "../store/index";
+import Station from "./Station";
+import { STATUSES } from "../constants/statuses";
+
+const ProductionLine = () => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  const handleAddVehicle = (key) => () =>
+    key === "1" ? dispatch({ type: "ADD_VEHICLE" }) : null;
+
+  const handleMoveVehicle = (key) => () =>
+    dispatch({ type: "MOVE_VEHICLE", current_station: parseInt(key) });
+
+  const handleVehicleStatusChange = (key) => (newStatus) => {
+    dispatch({
+      type: "UPDATE_VEHICLE_STATUS",
+      current_station: parseInt(key),
+      newStatus
+    });
+  };
+
+  const shouldWeShowMoveButton = (key, value) => {
+    if (
+      key === "4" ||
```

Comment:
💡 Cannot move a vehicle any further after it has already reached the Delivery Truck.